common/djangoapps/student/views.py

```diff
@@ -2486,18 +2486,22 @@ def validate_password(user, password):
         password: the user's proposed new password.
 
     Returns:
-        is_valid_password: a boolean indicating if the new password
-            passes the validation.
+        is_security_violated: a boolean indicating if the new password
+            passes the security policy validation.
         err_msg: an error message if there's a violation of one of the password
             checks. Otherwise, `None`.
     """
     err_msg = None
 
-    if settings.FEATURES.get('ENFORCE_PASSWORD_POLICY', False):
-        try:
+    try:
+        if settings.FEATURES.get('ENFORCE_PASSWORD_POLICY', False):
             validate_password_strength(password)
-        except ValidationError as err:
-            err_msg = _('Password: ') + '; '.join(err.messages)
+        else:
+            validate_password_length(password)
+
+    except ValidationError as err:
```

Comment:
@schenedx You can see `if` condition between both `validate_password_strength` and `validate_password_length`, so at one time only one method will be called and raise `ValidationError` which I believe is handled correctly.

```
    try:
        if settings.FEATURES.get('ENFORCE_PASSWORD_POLICY', False):
            validate_password_strength(password)
        else:
            validate_password_length(password)

    except ValidationError as err:
        # only strength/length is not valid, no security policy violated
        return None, _('Password: ') + '; '.join(err.messages)
```
I have confirmed that [validate_password_strength()](https://github.com/edx/edx-platform/blob/master/common/djangoapps/util/password_policy_validators.py#L29-L35) ultimately call different validators and all validators have their own message, which are excepted `as err` and not hard coded. So what ever message will be either because of strength violation or only length violation will be shown to user.

P.S: I have updated my comment, I believe which was creating confusion.